tools/test.py

```diff
@@ -138,6 +138,14 @@ def RunSingle(self, parallel, thread_id):
       try:
         start = datetime.now()
         output = case.Run()
+        # SmartOS has a bug that causes unexpected ECONNREFUSED errors.
+        # See https://smartos.org/bugview/OS-2767
+        # If ECONNREFUSED on SmartOS, retry the test one time.
+        if (output.UnexpectedOutput() and
+          sys.platform == 'sunos5' and 
+          'ECONNREFUSED' in output.output.stderr):
+            sys.stderr.write('ECONNREFUSED encountered on SmartOS; retrying test.')
```

Comment:
Sorry, but I think I was wrong about it. This sounds like a bad idea. We should fix it later.